src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java

```diff
@@ -0,0 +1,81 @@
+package uk.gov.ons.census.action.builders;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.action.model.UacQidTuple;
+import uk.gov.ons.census.action.model.entity.Case;
+import uk.gov.ons.census.action.model.entity.UacQidLink;
+import uk.gov.ons.census.action.model.repository.UacQidLinkRepository;
+
+@Component
+public class QidUacBuilder {
+  private static final int NUM_OF_UAC_IAC_PAIRS_NEEDED_BY_A_WALES_INITIAL_CONTACT_QUESTIONNAIRE = 2;
+  private static final int NUM_OF_UAC_IAC_PAIRS_NEEDED_FOR_SINGLE_LANGUAGE = 1;
+  private static final String WALES_IN_ENGLISH_QUESTIONNAIRE_TYPE = "02";
+  private static final String WALES_IN_WELSH_QUESTIONNAIRE_TYPE = "03";
+  public static final String HOUSEHLD_INITIAL_CONTACT_QUESTIONNIARE_TREATMENT_CODE_PREFIX = "HH_Q";
```

Comment:
Comment from another PR raised by @hbrace - "Feels like I'm being very picky but do we care about the misspelled QUESTIONNAIRE?"